src/Shared/TagTransformer.cs

```diff
@@ -113,27 +131,40 @@ private T TransformArrayTagInternal(string key, Array array, int? maxStringValue
             long[] => this.TransformArrayTag(key, array),
             float[] => this.TransformArrayTag(key, array),
             double[] => this.TransformArrayTag(key, array),
-            _ => this.ConvertToStringArrayThenTransformArrayTag(key, array, maxStringValueLength),
+            _ => this.ConvertToStringArrayThenTransformArrayTag(key, array, tagValueMaxLength),
         };
     }
 
-    private T ConvertToStringArrayThenTransformArrayTag(string key, Array array, int? maxStringValueLength)
+    private T ConvertToStringArrayThenTransformArrayTag(string key, Array array, int? tagValueMaxLength)
     {
-        string[] stringArray;
+        string?[] stringArray;
 
-        if (array is string[] arrayAsStringArray && (!maxStringValueLength.HasValue || !arrayAsStringArray.Any(s => s?.Length > maxStringValueLength)))
+        if (array is string?[] arrayAsStringArray
+            && (!tagValueMaxLength.HasValue || !arrayAsStringArray.Any(s => s?.Length > tagValueMaxLength)))
         {
             stringArray = arrayAsStringArray;
         }
         else
         {
-            stringArray = new string[array.Length];
+            stringArray = new string?[array.Length];
             for (var i = 0; i < array.Length; ++i)
             {
-                stringArray[i] = TruncateString(array.GetValue(i)?.ToString(), maxStringValueLength);
+                var item = array.GetValue(i);
+                stringArray[i] = item == null
```

Comment:
Is this check required? `TruncateString(array.GetValue(i)?.ToString(), maxStringValueLength);` would return `null` if the input string is `null` anyway, right?

Reply:
I think we do need it ya. I'm calling `Convert.ToString(object?)` which has a quirk in that if you give it `null` it returns `string.Empty`: https://github.com/dotnet/runtime/blob/8b04e76468af5905f739506e97039170b7554671/src/libraries/System.Private.CoreLib/src/System/Convert.cs#L1881

I say it is a quirk because I find this surprising:

```csharp
object? o = null;
Console.WriteLine(Convert.ToString(o) == null); // Writes False
		
string? s = null;
Console.WriteLine(Convert.ToString(s) == null); // Writes True

Console.WriteLine(Convert.ToString(null) == null); // Writes True
```

Reply:
So, you want to change the exporter behavior to export `null` instead of an empty string in those cases? Wouldn't this be a breaking change?

Reply:
Not exactly. The change here is a switch from calling `ToString` to calling `Convert.ToString`.

Before we had:

```csharp
stringArray[i] = TruncateString(array.GetValue(i)?.ToString(), maxStringValueLength);
```

Behavior: `null` values end up `null` in the array.

Now we have:

```csharp
                var item = array.GetValue(i);
                stringArray[i] = item == null
                    ? null
                    : TruncateString(Convert.ToString(item), tagValueMaxLength);
```

Behavior: `null` values end up `null` in the array.

Essentially this `null` check was added to not break the existing behavior.